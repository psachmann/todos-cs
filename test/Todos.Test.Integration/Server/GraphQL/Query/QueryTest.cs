namespace Todos.Test.Integration.Server.GraphQL;

public class QueryTest : IClassFixture<TodosServerFixture>
{
    private readonly TodosServerFixture _fixture;

    public QueryTest(TodosServerFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task GetTodoItem_ShouldReturnTodoItem()
    {
        var result = await _fixture.ExecuteQueryAsync((request) => request.SetQuery("{ todoItem { id, title, isDone } }"));

        result.Should().NotBeNull();
        result.MatchSnapshot();
    }
}
