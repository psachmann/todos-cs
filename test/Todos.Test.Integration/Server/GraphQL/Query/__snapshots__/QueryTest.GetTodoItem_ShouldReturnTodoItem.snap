{
  "data": {
    "todoItem": {
      "id": "97689b30-05d5-4b73-a21a-f4ef8f1cf3c1",
      "title": "Title #1",
      "isDone": false
    }
  }
}
